### PR TITLE
[C API DOC][MIGRATION] Add C API migration info from 1.0 to 2.0

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Add C API migration info from 1.0 to 2.0
 - Add pipeline migration
 - Add device config migration
 - Add preprocessing migration

### Tickets:
 - 93265

### CODE REVIEW

**Comment by zhaixuejun1993** [comment]
@riverlijunjie  @ilyachur  Pls help review.

File: docs/snippets/ov_common.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    //! [ov_api_2_0:compile_model]
+    ov_compiled_model_t* compiled_model = NULL;
+    ov_core_compile_model(core, model, "CPU", 0, &compiled_model);
+    //! [ov_api_2_0:compile_model]
+
+    //! [ov_api_2_0:create_infer_request]
+    ov_infer_request_t* infer_request = NULL;
+    ov_compiled_model_create_infer_request(compiled_model, &infer_request);
+    //! [ov_api_2_0:create_infer_request]
+
```
Comment (riverlijunjie) [inline docs/snippets/ov_common.c:115]: `inputs_aligned()` is missing.

File: docs/snippets/ov_preprocessing_migration.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    ov_preprocess_input_model_info_t* input_model = NULL;
+    ov_layout_t* layout = NULL;
+    ov_model_t* new_model = NULL;
+
+    ov_preprocess_prepostprocessor_create(model, &preprocess);
+    ov_preprocess_prepostprocessor_get_input_info_by_index(preprocess, 0, &input_info);
+    ov_preprocess_input_info_get_model_info(input_info, &input_model);
+    // we only need to know where is C dimension
+    ov_layout_create("...C", &layout);
+    ov_preprocess_input_model_info_set_layout(input_model, layout);
```
Comment (riverlijunjie) [inline docs/snippets/ov_preprocessing_migration.c:29]: In C++ version, here is:
`input.preprocess().mean(116.78f).scale({ 57.21f, 57.45f, 57.73f });`
But C 2.0 API doesn't provide mean() and scale() interface, do we need add such interfaces?

**Comment by ilyachur** [comment]
> @ilyachur Considered that we have no C 1.0 API document before, but for C API migration we need it, so do we need add some C 1.0 API as this PR has done to demonstrate C API migration or just set empty for C API 1.0?

Yes this PR contains the right idea we have a file for 1.0 API and show how this code can be rewritten to OV 2.0

File: docs/snippets/ov_preprocessing_migration.c (hunk 2 above)
Comment (ilyachur) [reply docs/snippets/ov_preprocessing_migration.c:29]: As I see we also don't have:

- PreProcessSteps& reverse_channels();
- PreProcessSteps& convert_layout();
- PreProcessSteps& crop();
- PreProcessSteps& scale();
- PreProcessSteps& mean();

Honestly I am not sure that anybody will use it, but if we would like to have a full picture - we need to add all these methods. We already have:

- PreProcessSteps& convert_element_type();
- PreProcessSteps& convert_color();
- PreProcessSteps& resize();

**Comment by ilyachur** [comment]
I re-run validation because I added `Code snippets / Build` steps which allow to check changes in snippets for Windows, Linux and Mac

**Review by ilyachur — APPROVED** [review]

**Pull request merged**